packages/dd-trace/src/profiling/profiler.js

```diff
@@ -218,7 +243,14 @@ class Profiler extends EventEmitter {
       // encode and export asynchronously
       for (const { profiler, profile } of profiles) {
         try {
-          encodedProfiles[profiler.type] = await profiler.encode(profile)
+          const encoded = await profiler.encode(profile)
+          let compressed
+          if (encoded instanceof Buffer && this._compressionFn !== undefined) {
```

Comment:
The test on the type of `encoded` is there for unit tests ?

Reply:
yes :-)